### PR TITLE
Simplify Gallery Setup

### PR DESCRIPTION
**Pull request by loic-sharma**
Simplified Gallery Setup:

* Added Visual Studio 2017 and PowerShell 4.0 requirements
* Removed `nuget.localtest.me`, we will use `localhost` now
* Renamed `Enable-LocalTestMe.ps1` to `Setup-DevEnvironment.ps1`
    * `Setup-DevEnvironment.ps1` no longer requires Windows SDK
    * `Setup-DevEnvironment.ps1` runs migrations.

The new setup procedure is:

1. Install Visual Studio 2017 with ASP.NET and Azure workloads
2. Run the `build` command
3. Run the `Setup-DevEnvironment` tool

That's it! 😄

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -9,7 +9,7 @@ param (
     [string]$SemanticVersion = '1.0.0-zlocal',
     [string]$Branch,
     [string]$CommitSHA,
-    [string]$BuildBranch = '1c479a7381ebbc0fe1fded765de70d513b8bd68e'
+    [string]$BuildBranch = 'ed0316f287f23a48b0b9b7909eadd62b0c73aa73'
```
Comment (loic-sharma) [inline build.ps1]: __NOTE__: I will have to update this hash once Scott updates ServerCommon's master branch.

Comment (scottbommarito) [inline build.ps1]: 37ff6e758c38b3f513af39f881399ce85f4ff20b

Comment (loic-sharma) [inline build.ps1]: Done!

File: README.md [hunk 2]
```diff
@@ -136,13 +98,13 @@ This is the Git workflow we're currently using:
...
 2.  __Create a topic branch to do your work.__
-    You must work in topic branches, in order to help us keep our features isolated and easily moved between branches.
+    You must work in topic branches to help us keep our features isolated and easily moved between branches.
     Our policy is to start all topic branches off of the 'dev' branch. 
     Branch names should use the following format '[user]-[bugnumber]-[shortdescription]'. If there is no bug yet, 
     create one and assign it to yourself!
 
         git checkout dev
-        git checkout -b anurse-123-makesuckless
+        git checkout -b anurse-123-improve
```
Comment (joelverhagen) [inline README.md]: hahahahaha

File: README.md [hunk 3]
```diff
@@ -10,79 +10,42 @@ This project has adopted the [Microsoft Open Source Code of Conduct](https://ope
...
+You can also turn off e-email confirmations by changing the value of `Gallery.ConfirmEmailAddresses` to `false`
+in the `src\NugetGallery\Web.Config` file.
+
+Visual Studio may modify the `applicationhost.config` file. You can force git to ignore changes to this file
+with:
 
     git update-index --assume-unchanged .vs/config/applicationhost.config
 
-You can undo this with this command:
+You can undo this with this command with:
```
Comment (xavierdecoster) [inline README.md]: Sounds like one `with` too many :)

File: README.md (hunk 2 above)
Comment (xavierdecoster) [inline README.md]: Lol :) Should we perhaps replace with a more generic `username-issuenumber`? Or is this a real sample?

Comment (joelverhagen) [inline README.md]: Yes, agreed.

File: build.ps1 [hunk 4]
```diff
@@ -9,7 +9,7 @@ param (
     [string]$SemanticVersion = '1.0.0-zlocal',
     [string]$Branch,
     [string]$CommitSHA,
-    [string]$BuildBranch = '1c479a7381ebbc0fe1fded765de70d513b8bd68e'
+    [string]$BuildBranch = '37ff6e758c38b3f513af39f881399ce85f4ff20b'
 )
 
 # For TeamCity - If any issue occurs, this script fail the build. - By default, TeamCity returns an exit code of 0 for all powershell scripts, even if they fail
```
Comment (skofman1) [inline build.ps1]: We no longer use TeamCity. Can this be removed?

File: build.ps1 [hunk 5]
```diff
@@ -9,7 +9,7 @@ param (
     [string]$SemanticVersion = '1.0.0-zlocal',
     [string]$Branch,
     [string]$CommitSHA,
-    [string]$BuildBranch = '1c479a7381ebbc0fe1fded765de70d513b8bd68e'
+    [string]$BuildBranch = '37ff6e758c38b3f513af39f881399ce85f4ff20b'
```
Comment (skofman1) [inline build.ps1]: what is this id?

File: src/NuGetGallery/Web.config [hunk 6]
```diff
@@ -51,7 +51,7 @@
     <!-- Set this if you have a Google Analytics property for the site -->
     <add key="Gallery.AzureCdnHost" value="" />
     <!-- Set this to the Azure CDN Host you are using for blob storage, if you have configured one -->
-    <add key="Gallery.SiteRoot" value="http://nuget.localtest.me/" />
+    <add key="Gallery.SiteRoot" value="http://localhost" />
```
Comment (skofman1) [inline src/NuGetGallery/Web.config]: https?

File: build.ps1 (hunk 5 above)
Comment (loic-sharma) [inline build.ps1]: This is the [latest commit](https://github.com/NuGet/ServerCommon/commit/37ff6e758c38b3f513af39f881399ce85f4ff20b) on ServerCommon's master branch.

File: src/NuGetGallery/Web.config (hunk 6 above)
Comment (loic-sharma) [inline src/NuGetGallery/Web.config]: It seems like the protocol used in the `Gallery.SiteRoot` doesn't matter as the `ConfigurationService` exposes the [`GetSiteRoot`](https://github.com/NuGet/NuGetGallery/blob/master/src/NuGetGallery/Configuration/ConfigurationService.cs#L73) method that will re-map the URL to HTTP/HTTPS depending on a boolean input.

File: build.ps1 (hunk 4 above)
Comment (loic-sharma) [inline build.ps1]: I think this is good to have as it catches uncaught exceptions and dumps them to the output. I'll update the comment though!

File: build.ps1 [hunk 7]
```diff
@@ -9,10 +9,10 @@ param (
...
 )
 
-# For TeamCity - If any issue occurs, this script fail the build. - By default, TeamCity returns an exit code of 0 for all powershell scripts, even if they fail
+# This script should fail the build if any issue occurs.
```
Comment (scottbommarito) [inline build.ps1]: If we want to change this, we should change it for all our build scripts and also common.ps1

**Pull request merged**